robot_framework/process.py

```diff
@@ -1,8 +1,96 @@
 """This module contains the main process of the robot."""
 
+import os
+import json
+from datetime import datetime
+import re
+
+import pyodbc
 from OpenOrchestrator.orchestrator_connection.connection import OrchestratorConnection
+from itk_dev_shared_components.kmd_nova.authentication import NovaAccess
+from itk_dev_shared_components.graph import authentication as graph_authentication
+from itk_dev_shared_components.graph import mail as graph_mail
+from itk_dev_shared_components.smtp import smtp_util
+
+from robot_framework.sub_process import indkomst, database, nova
+from robot_framework import config
 
 
 def process(orchestrator_connection: OrchestratorConnection) -> None:
     """Do the primary process of the robot."""
     orchestrator_connection.log_trace("Running process.")
+
+    graph_creds = orchestrator_connection.get_credential(config.GRAPH_API)
+    graph_access = graph_authentication.authorize_by_username_password(graph_creds.username, **json.loads(graph_creds.password))
+
+    mails = graph_mail.get_emails_from_folder("itk-rpa@mkb.aarhus.dk", "Indbakke/Udrejsekontrol", graph_access)
+    mails = [mail for mail in mails if mail.sender == 'noreply@aarhus.dk' and mail.subject == 'RPA - Udrejsekontrol (fra Selvbetjening.aarhuskommune.dk)']
+    mails.sort(key=lambda m: datetime.fromisoformat(m.received_time))
+
+    if not mails:
+        orchestrator_connection.log_info("No emails in queue.")
+        return
+
+    for mail in mails:
+        email_text = mail.get_text()
+        sender_email = re.findall("BrugerE-mail: (.+?)AZ-ident", email_text)[0]
+        sender_ident = re.findall("AZ-ident: (.+?)Antal", email_text)[0]
+        requested_count = int(re.findall(r"Antal ønskede sager(\d+)", email_text)[0])
+
+        approved_senders = json.loads(orchestrator_connection.process_arguments)["approved_senders"]
+
+        if sender_ident not in approved_senders:
+            orchestrator_connection.log_info(f"Request denied for: {sender_email} - {sender_ident}")
+            smtp_util.send_email(sender_email, "itk-rpa@mkb.aarhus.dk", "Anmodning afvist", "Din anmodning til Udrejsekontrol er blevet afvist, da du ikke er på listen af godkendt medarbejdere.", smtp_server=config.SMTP_SERVER, smtp_port=config.SMTP_PORT)
```

Comment:
Der mangler et bogstav
```suggestion
            smtp_util.send_email(sender_email, "itk-rpa@mkb.aarhus.dk", "Anmodning afvist", "Din anmodning til Udrejsekontrol er blevet afvist, da du ikke er på listen af godkendte medarbejdere.", smtp_server=config.SMTP_SERVER, smtp_port=config.SMTP_PORT)
```